46.md

```diff
@@ -0,0 +1,159 @@
+NIP-46
+======
+
+Nostr Connect
+------------------------
+
+`draft` `optional` `author:tiero` `author:giowe` `author:vforvalerio87`
+
+## Rationale
+
+Private keys should be exposed to as few systems - apps, operating systems, devices - as possible as each system adds to the attack surface.
+
+Entering private keys can also be annoying and requires exposing them to even more systems such as the operating system's clipboard that might be monitored by malicious apps.
+
+
+## Terms
+
+* **App**: Nostr app on any platform that *requires* to act on behalf of a nostr account.
+* **Signer**: Nostr app that holds the private key of a nostr account and *can sign* on its behalf.
+
+
+## `TL;DR`
+
+
+**App** and **Signer** sends ephemeral encrypted messages to each other using kind `24133`, using a relay of choice. 
+
+App prompts the Signer to do things such as fetching the public key or signing events.
+
+The `content` field must be an encrypted JSONRPC-ish **request** or **response**.
+
+## Signer Protocol
+
+### Messages
+
+#### Request
+
+```json
+{
+  "id": <random_string>,
+  "method": <one_of_the_methods>,
+  "params": [<anything>, <else>]
+}
+```
+
+#### Response
+
+```json
+{
+  "id": <request_id>,
+  "result": <anything>,
+  "error": <reason>
+}
+```
+
+### Methods
+
+
+#### Mandatory 
+
+These are mandatory methods the remote signer app MUST implement:
+
+- **get_public_key**
+  - params []
+  - result `pubkey` 
+- **sign_event**
+  - params [`event`]
+  - result `signature` 
+
+#### optional
```

Comment:
I think it would be nice to be able to query all supported methods.

Reply:
Good point. I don't know if makes sense to be a mandatory method?

Reply:
I think it's a good idea to have it as a mandatory method. Because it will be then always possible, to ask for the supported methods.

Reply:
Will do

Reply:
done